workspaces/servicenow/plugins/servicenow-backend/src/service/router.ts

```diff
@@ -68,5 +68,54 @@ export async function createRouter(
     }
   });
 
+  router.get('/cmdb/business-application/:appcode', async (req, res) => {
+    const appCode = req.params.appcode;
+    try {
+      const incidents = await client.getBusinessApplication(appCode);
+      res.json(incidents);
+    } catch (error) {
+      if (error instanceof InputError) {
+        throw error;
+      }
+
+      // Log the full error and throw a generic one
+      logger.error(
+        'Failed to fetch business-application from ServiceNow',
+        error,
+      );
+      throw new Error('Failed to fetch business-application from ServiceNow');
+    }
+  });
+  router.get('/cmdb/user/:userid', async (req, res) => {
+    const userId = req.params.userid;
+    try {
+      const incidents = await client.getUserDetails(userId);
+      res.json(incidents);
+    } catch (error) {
+      if (error instanceof InputError) {
+        throw error;
+      }
```

Comment:
This gets "critical" (personal) user information from ServiceNow? Do you think we should add extra permissions checks to this plugin for this?